### DISCUSSION TITLE
ccache support for all builds

### DISCUSSION THREAD

**Discussion by Tagman** [microsoft/vcpkg#26032] [Q&A] [answered]
Hi all,
is there a way to get ccache to work with using the manifest mode.
Ideally ccache should be used in every package built.

I tried using `-DRULE_LAUNCH_COMPILE=ccache` on my root cmake call like:
`cmake -S C:\sources -B C:\build -DCMAKE_TOOLCHAIN_FILE=C:\vcpkg\scripts\buildsystems\vcpkg.cmake -DVCPKG_VERBOSE=ON -DRULE_LAUNCH_COMPILE=ccache`

I also tried to set the `CC` and `CXX` environment variables to `ccache`.
 I even tried to pass `-DRULE_LAUNCH_COMPILE=ccache` as a vcpkg_cmake_configure OPTIONS.

Any ideas why this is not working are apprecciated

**Comment by Osyotr** [comment]
`RULE_LAUNCH_COMPILE` is for CTest, try `CMAKE_CXX_COMPILER_LAUNCHER`

**Comment by Tagman** [comment]
Sadly this also did not work. I've set `CMAKE_CXX_COMPILER_LAUNCHER` as an environment variable, in the portfile of the port and when calling my root cmake script. Also tried `CXX_COMPILER_LAUNCHER` with the same results. 

`ccache -s` showed no bytes saved.
I made sure in between the run to delete all previously built part of the port, so that it always started a recompile.

**Reply by Neumann-A** [reply]
if you are using the windows triplets setting env variables will do nothing unless you forward them

**Reply by Tagman** [reply]
Thanks for that idea but it is still not working. 
I've set the environment varable `VCPKG_KEEP_ENV_VARS` to `Path;CMAKE_CXX_COMPILER_LAUNCHER`

**Reply by Neumann-A** [reply]
> I've set the environment varable VCPKG_KEEP_ENV_VARS

That is a triplet var not a env var.

**Reply by autoantwort** [reply]
It is an env var: https://learn.microsoft.com/en-us/vcpkg/users/config-environment#vcpkg_keep_env_vars

**Comment by Tagman** [comment, marked as answer]
I got it to work with creating symlinks inside the ccache folder.

`mklink C:\ccache\cl.exe .\ccache.exe`
this creates a cl.exe file that links to ccache.exe
I then set the ccache folder in my Path environment variable. Making sure it is located before the path to the original cl.exe

It doesn't feel clean, but it gets the job done.
